app/models/customer_resource.rb

```diff
@@ -20,9 +20,37 @@ class CustomerResource < RmApiResource
 
   before_request do |name, request|
     if name == :find_by_package
-      request.get_params[:search] = ''
+      filters = request.get_params.delete(:filter) || {}
+
+      unless filters.is_a?(ActionController::Parameters) || filters.is_a?(Hash)
```

Comment:
Is this necessary? As far as I can tell, this is cranked through the `**` operator here, so I don't see how this could ever be anything but a `Hash`

Reply:
`filter` is a named param. You can pass `?q=search&filter=blah` and it would work up until we try to pull a value off of `filter`. This was first introduced when `filter` briefly did accept a string instead of a hash.

Reply:
You are correct in that it is 'munging' params. The `params_for_rmapi` may work. But does it matter that what params are passed to RMAPI is dependent upon wether its a Package, Package-Title?